File: LeetCode-Practise/Valid_Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dict1 = {}
        dict2 = {}
        if len(s) == len(t) and set(s)==set(t):
            for element in set(s):
                dict1[element] = s.count(element)
                dict2[element] = t.count(element)
            if dict1 == dict2:
                return True
            return False
        return False

File: valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()){
            return false;
        }
        
        int[] freq = new int[26];
        for(int i=0; i<s.length(); i++){
            freq[s.charAt(i) - 'a']++;
            freq[t.charAt(i) - 'a']--;
        }
        
        for(int i=0; i<freq.length; i++){
            if(freq[i] != 0){
                return false;
            }
        }
        return true;
    }
}